Mathlib/Data/Matroid/Minor/Basic.lean

```diff
@@ -716,4 +716,197 @@ lemma IsCocircuit.delete_diff_isCocircuit {X : Set α} (hK : M.IsCocircuit K) (h
+
+lemma contract_delete_contract_delete (M : Matroid α) (C D C' D' : Set α) (h : Disjoint C' D) :
+    M ／ C ＼ D ／ C' ＼ D' = M ／ (C ∪ C') ＼ (D ∪ D') := by
+  rw [contract_delete_contract_delete', sdiff_eq_left.mpr h]
+
+/-- A version of `delete_contract_delete` without the disjointness hypothesis,
+and hence a less simple RHS. -/
+lemma delete_contract_delete' (M : Matroid α) (D C D' : Set α) :
+    M ＼ D ／ C ＼ D' = M ／ (C \ D) ＼ (D ∪ D') := by
+  rw [delete_contract_comm', delete_delete]
+
+lemma delete_contract_delete (M : Matroid α) (D C D' : Set α) (h : Disjoint C D) :
+    M ＼ D ／ C ＼ D' = M ／ C ＼ (D ∪ D') := by
+  rw [delete_contract_delete', sdiff_eq_left.mpr h]
+
+/-! ### Minors -/
+
+/-- `N` is a minor of `M` if `N = M ／ C ＼ D` for some `C` and `D`.
+The definition itself does not require `C` and `D` to be disjoint,
+or even to be subsets of the ground set. See `Matroid.IsMinor.exists_eq_contract_delete_disjoint`
+for the fact that we can choose `C` and `D` with these properties. -/
+def IsMinor (N M : Matroid α) : Prop := ∃ C D, N = M ／ C ＼ D
+
+/-- `≤m` denotes the minor relation on matroids. -/
+infixl:50 " ≤m " => Matroid.IsMinor
+
+@[simp]
+lemma contract_delete_isMinor (M : Matroid α) (C D : Set α) : M ／ C ＼ D ≤m M :=
+  ⟨C, D, rfl⟩
+
+lemma IsMinor.exists_eq_contract_delete_disjoint (h : N ≤m M) :
+    ∃ (C D : Set α), C ⊆ M.E ∧ D ⊆ M.E ∧ Disjoint C D ∧ N = M ／ C ＼ D := by
+  obtain ⟨C, D, rfl⟩ := h
+  exact ⟨C ∩ M.E, (D ∩ M.E) \ C, inter_subset_right, diff_subset.trans inter_subset_right,
+    disjoint_sdiff_right.mono_left inter_subset_left,
+    by simp [delete_eq_delete_iff, inter_assoc, inter_diff_assoc]⟩
+
+/-- `N` is a strict minor of `M` if `N` is a minor of `M` and `N ≠ M`.
+Equivalently, `N` is obtained from `M` by deleting/contracting subsets of the ground set
+that are not both empty. -/
+def IsStrictMinor (N M : Matroid α) : Prop := N ≤m M ∧ ¬ M ≤m N
+
+/-- `<m` denotes the strict minor relation on matroids. -/
+infixl:50 " <m " => Matroid.IsStrictMinor
+
+lemma IsMinor.subset (h : N ≤m M) : N.E ⊆ M.E := by
+  obtain ⟨C, D, rfl⟩ := h
+  exact diff_subset.trans diff_subset
+
+lemma IsMinor.refl {M : Matroid α} : M ≤m M := ⟨∅, ∅, by simp⟩
+
+lemma IsMinor.trans {M₁ M₂ M₃ : Matroid α} (h : M₁ ≤m M₂) (h' : M₂ ≤m M₃) : M₁ ≤m M₃ := by
+  obtain ⟨C₁, D₁, rfl⟩ := h
+  obtain ⟨C₂, D₂, rfl⟩ := h'
+  exact ⟨C₂ ∪ C₁ \ D₂, D₂ ∪ D₁, by rw [contract_delete_contract_delete']⟩
+
+lemma IsMinor.eq_of_ground_subset (h : N ≤m M) (hE : M.E ⊆ N.E) : M = N := by
+  obtain ⟨C, D, rfl⟩ := h
+  rw [delete_ground, contract_ground, subset_diff, subset_diff] at hE
+  rw [← contract_inter_ground_eq, hE.1.2.symm.inter_eq, contract_empty, ← delete_inter_ground_eq,
+    hE.2.symm.inter_eq, delete_empty]
+
+lemma IsMinor.antisymm (h : N ≤m M) (h' : M ≤m N) : N = M :=
+  h'.eq_of_ground_subset h.subset
+
+/-- The minor order is a `PartialOrder` on `Matroid α`.
+We prefer the spelling `N ≤m M` over `N ≤ M` for the dot notation. -/
```

Comment:
I don't know enough about matroids... but it sounds like you are now making it somewhat inconvenient to access  a whole bunch of partial order API. Especially automation will trip over this boundary.
But I can't weigh the costs. I don't know how inconvenient loss of dot-notation would be.

Reply:
The analogy I had in mind was sets and subsets, where there is a custom notation. The use of dot notation for minors will be heavier than what it is for subsets. 

In practice, it is not so common to be doing matroid theory at the level of abstract orderings, and it is very common to say 'let `C` and `D` be contract/delete sets' after you have a minor. There is also the fact that there are at least two reasonable partial orders on `Matroid a`: the minor order `≤m` and the restriction order `≤r`. These are closely related, and imo it breaks the natural expectations if dot notation is idiomatic for one but not the other.

Reply:
Ok, let's try this out. I'll gladly defer to your expertise.